dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -0,0 +1,180 @@
+    JavaClass javaClass = parser.parse();
+    return listConstantPoolMethodRef(javaClass);
+  }
+
+  /**
+   *  Lists the methodref entries owned by the defined class in class file
+   *
+   * @param classFileStream stream of a class file
+   * @param fileName name of the file that contains class
+   * @return methodref entries owned by the class
+   * @throws IOException when there is problem in reading classFileStream
+   */
+  public static List<ConstantPoolMethodref> listOwningConstantPoolMethodref(InputStream classFileStream,
+      String fileName) throws IOException {
+    ClassParser parser = new ClassParser(classFileStream, fileName);
+    JavaClass javaClass = parser.parse();
+    List<ConstantPoolMethodref> methodrefs = listConstantPoolMethodRef(javaClass);
+    Set<String> owningClass = Sets.newHashSet(javaClass.getClassName());
+    List<ConstantPoolMethodref> owningMethodrefs =  methodrefs
+            .stream()
+            .filter(methodref -> owningClass.contains(methodref.getClassName()))
+            .collect(Collectors.toList());
+    return owningMethodrefs;
+  }
+
+  /**
+   *  Lists the methodref entries not owned by the defined class in class file
+   *  The list the class attemts to invoke the methods defined elsewhere
+   *
+   * @param classFileStream stream of a class file
+   * @param fileName name of the file that contains class
+   * @return methodref entries external to the class
+   * @throws IOException when there is problem in reading classFileStream
+   */
+  public static List<ConstantPoolMethodref> listExternalConstantPoolMethodref(
+      InputStream classFileStream, String fileName) throws IOException {
+    ClassParser parser = new ClassParser(classFileStream, fileName);
+    JavaClass javaClass = parser.parse();
+    List<ConstantPoolMethodref> methodrefs = listConstantPoolMethodRef(javaClass);
+    Set<String> owningClass = Sets.newHashSet(javaClass.getClassName());
+    List<ConstantPoolMethodref> owningMethodrefs =
+        methodrefs
+            .stream()
+            .filter(methodref -> !owningClass.contains(methodref.getClassName()))
+            .collect(Collectors.toList());
+    return owningMethodrefs;
+  }
+
+  /**
+   * Lists methods entries defined in the class file
+   *
+   * @param classFileStream stream of a class file
+   * @param fileName name of the file that contains class
+   * @return method and signature entries defined in the class file
+   * @throws IOException when there is problem in reading classFileStream
+   */
+  public static List<MethodAndSignature> listDeclaredMethods(InputStream classFileStream, String fileName)
+      throws IOException {
+    final ClassParser parser = new ClassParser(classFileStream, fileName);
+    final JavaClass javaClass = parser.parse();
+    Method[] methods = javaClass.getMethods();
+
+    List<MethodAndSignature> signatures = Arrays.stream(methods)
+        .map(method -> new MethodAndSignature(method.getName(), method.getSignature()))
```

Comment:
So it appears that BCEL uses non-standard terminology. Here is Oracle's definition:

https://docs.oracle.com/javase/tutorial/java/javaOO/methods.html

"method signature" is supposed to be the combination of the method name and the parameters, it isn't just supposed to be the parameter types (as BCEL seems to use "signature" for). I'd prefer our tool to use the correct terminology. That would imply `MethodSignature` for the class name instead of `MethodAndSignature`.